src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -136,31 +136,48 @@ private static MsQuicApi AllocateMsQuicApi()
     {
         Debug.Assert(IsQuicSupported);
 
+        int openStatus = MsQuic.QUIC_STATUS_INTERNAL_ERROR;
+
         if (TryLoadMsQuic(out IntPtr msQuicHandle) &&
-            TryOpenMsQuic(msQuicHandle, out QUIC_API_TABLE* apiTable))
+            TryOpenMsQuic(msQuicHandle, out QUIC_API_TABLE* apiTable, out openStatus))
         {
             return new MsQuicApi(apiTable);
         }
 
+        ThrowHelper.ThrowIfMsQuicError(openStatus);
+
+        // this should unreachable as TryOpenMsQuic returns non-success status on failure
         throw new Exception("Failed to create MsQuicApi instance");
```

Comment:
If you were to invert this condition, you wouldn't need the unreachable throw:
```C#
if (!TryLoadMsQuic(out IntPtr msQuicHandle) || !TryOpenMsQuic(msQuicHandle, out QUIC_API_TABLE* apiTable, out openStatus))
{
    ThrowHelper.ThrowIfMsQuicError(openStatus);
}

return new MsQuicApi(apiTable);
```
You could also have a ThrowMsQuicError that was unconditional. Or just use `throw GetExceptionForMsQuicStatus(openStatus);`.